pynestml/codegeneration/printers/gsl_variable_printer.py

```diff
@@ -82,10 +82,11 @@ def _print_buffer_value(self, variable: ASTVariable) -> str:
         variable_symbol = variable.get_scope().resolve_to_symbol(variable.get_complete_name(), SymbolKind.VARIABLE)
         if variable_symbol.is_spike_input_port():
             var_name = variable_symbol.get_symbol_name().upper()
-            if variable_symbol.get_vector_parameter() is not None:
-                vector_parameter = ASTUtils.get_numeric_vector_size(variable_symbol)
-                var_name = var_name + "_" + str(vector_parameter)
-
+            if variable.has_vector_parameter():
+                if variable.get_vector_parameter().is_variable():
+                    var_name += "_0 + " + variable.get_vector_parameter().get_variable().get_name()
```

Comment:
why the 0 here?

Reply:
Because the enum corresponding to the first input port in a vector of input ports will have the `_0` suffixed to the enum's name. For example:

```
input:
  foo[42] <- spike

update:
  j integer = foo[23]
```
The enum for the first input port here would be `FOO_0`. So the indexing in this case would be

```
int j = spike_inputs_buffer[FOO_0 + 23]
```